### DISCUSSION TITLE
ELF Executor?

### DISCUSSION THREAD

**Discussion by NullDev** [shiro-os/shiro-kernel#4] [Q&A] [answered]
Should we build an [ELF Shellcode Executor](https://stackoverflow.com/questions/27900201/create-and-test-x86-64-elf-executable-shellcode-on-a-linux-machine) or should we rely on our own Process Images?

**Comment by CreepSore** [comment, marked as answer]
To be honest, an ELF Loader would be pretty neat. It's on a whole other level than making something own though, because you still have to provide the Kernel bindings. However, once it's done, we could actually use an already existing JavaScript Framework like NodeJS, and we don't have to rely on our own.

I think, for now we're good with our own, but after setting up some of the most important kernel bindings, we can try making an ELF Loader a priority.

**Reply by NullDev** [reply]
Awesome! Thanks :)
